apps/ocean-api/__tests__/controllers/RawTxController.test.ts

```
@@ -1,40 +1,118 @@
-  expect(feeRate).not.toStrictEqual(0.00005000)
+    const feeRate = await apiTesting.client.rawtx.feeEstimate()
+    expect(feeRate).not.toStrictEqual(0.00005000)
+  })
+
+  it('should be able to provide confirmation target', async () => {
+    const feeRate = await apiTesting.client.rawtx.feeEstimate(100)
+
+    expect(feeRate).toBeDefined()
+  })
 })
 
-it('should be able to provide confirmation target', async () => {
-  const feeRate = await apiTesting.client.rawtx.feeEstimate(100)
+describe('test', () => {
+  beforeAll(async () => {
+    await apiTesting.start()
+    await apiTesting.testing.container.waitForWalletCoinbaseMaturity()
+  })
+
+  afterAll(async () => {
+    await apiTesting.stop()
+  })
+
+  it('should accept valid txn', async () => {
+    const { hex: { signed } } = await apiTesting.testing.rawtx.fund({
+      a: { amount: 10, ellipticPair: privA },
+      b: { amount: 9.9999, ellipticPair: privB }
+    })
 
-  expect(feeRate).toBeDefined()
+    await apiTesting.client.rawtx.test({
+      hex: signed
+    })
+  })
+
+  it('should accept valid txn with given maxFeeRate', async () => {
+    const { hex: { signed } } = await apiTesting.testing.rawtx.fund({
+      a: { amount: 10, ellipticPair: privA },
+      b: { amount: 9.9999, ellipticPair: privB }
+    })
+    await apiTesting.client.rawtx.test({
+      hex: signed,
+      maxFeeRate: 0.05
+    })
+  })
+
+  it('should reject due to invalid txn', async () => {
+    expect.assertions(2)
+    try {
+      await apiTesting.client.rawtx.test({ hex: '0400000100881133bb11aa00cc' })
+    } catch (err: any) {
+      expect(err).toBeInstanceOf(ApiException)
+      expect(err.error).toStrictEqual({
+        code: 400,
+        type: 'BadRequest',
+        message: '400 - BadRequest  : Transaction decode failed',
+        at: expect.any(Number),
+        url: '/v1/regtest/rawtx/test'
+      })
+    }
+  })
+
+  it('should reject due to high fees', async () => {
```

question: is this test added as ain client no longer supports a boolean?

---

hmm... i don't quite get it.. support boolean???

---

nope.. diff.. the test resulted the fee was greater the `maxFeeRate: 1` set, hence error
if assign bool on 2nd params should throw the error you stated above